File: KLASA_3/ROZDZIAL_1/LEKCJA_1/PYTHON/ZAD9.py
from stos import Stos


def sortuj(lista, sposob_sortowania):
    stos=Stos()
    stos_temp=Stos()
    for i in range(len(lista)):
        while (not stos.isEmpty()) and (
                stos.peek() <= lista[i] if sposob_sortowania == 0
                else stos.peek() >= lista[i]
        ):
            stos_temp.push(stos.peek())
            stos.pop()
        stos.push(lista[i])
        while (not stos_temp.isEmpty()):
            stos.push(stos_temp.peek())
            stos_temp.pop()
    return stos


liczby=[int(x) for x in input('Podaj liczby: ').split()]
jak_sortowac=int(input('Jak sortować? 0 - malejąco, 1 - rosnąco: '))
print(sortuj(liczby, jak_sortowac))

File: KLASA_3/ROZDZIAL_1/LEKCJA_1/C++/ZAD9.cpp
#include <iostream>
#include <stack>
#include <vector>
#include <sstream>

using namespace std;

stack<int> sortuj(vector<int> liczby, int sposob_sortowania)
{
    stack<int> stos;
    stack<int> stos_temp;

    for(int i=0;i<liczby.size();i++){
        while(!stos.empty() && (
                (sposob_sortowania == 1) ? stos.top() <= liczby[i] : stos.top() >= liczby[i]
        )){
            stos_temp.push(stos.top());
            stos.pop();
        }
        stos.push(liczby[i]);
        while (!stos_temp.empty()){
            stos.push(stos_temp.top());
            stos_temp.pop();
        }
    }

    return stos;
}

int main()
{
    string liczby_str;
    cout << "Podaj liczby: ";
    getline(cin, liczby_str);
    stringstream iss(liczby_str);

    vector<int> liczby;

    int liczba;
    while (iss >> liczba) {
        liczby.push_back(liczba);
    }

    int sposob;
    cout << "Jak sortowac? 0 - malejaco, 1 - rosnaco: ";
    cin >> sposob;

    stack<int> posortowane = sortuj(liczby, sposob);

    while(!posortowane.empty()){
        cout << posortowane.top() << " ";
        posortowane.pop();
    }

    return 0;
}
